### PR TITLE
Add initial ycsb support

### PR DESCRIPTION
**Pull request by calderast**
Add initial support for the Yahoo! Cloud Serving Benchmark (YCSB) with Azure Cosmos DB.

### CODE REVIEW

File: perf/ycsb/build.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	<!-- set global properties for this build -->
+	<property name="DEST" value="${BUILD_ROOT}/perf/ycsb" />
+	<property name="SRC" location="." />
+	
+	<target name="init">
+		<mkdir dir="${DEST}" />
+	</target>
+
+	<target name="getYCSB" depends="init">
+		<var name="TAR_NAME" value="ycsb-0.17.0.tar.gz"/>
```
Comment (karianna) [inline perf/ycsb/build.xml]: can we extra the version as a property?

File: perf/ycsb/build.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	<target name="dist" depends="getYCSB" description="generate the distribution">
+	    <gunzip src="ycsb-0.17.0.tar.gz"/>
+		<untar src="ycsb-0.17.0.tar" dest="${DEST}"/>
+	</target>
+
+	<target name="getPropFiles" depends="dist">
+		<exec executable="bash">
+            <arg value="${SRC}/getPropertiesFiles.sh"/>
+			<arg value="${DEST}"/>
+		</exec>
```
Comment (karianna) [inline perf/ycsb/build.xml:59]: whitespace weirdness - maybe check whole doc

File: perf/ycsb/getPropertiesFiles.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+true=0
+false=1
+
+propFilesExist() {
```
Comment (karianna) [inline perf/ycsb/getPropertiesFiles.sh:8]: needs some documentation.  Has shellcheck been run?  Should we have set -ev?

File: perf/ycsb/getPropertiesFiles.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    else 
+        echo "No properties files found."
+        return $false
+    fi 
+}
+
+ycsb_dir=$1
+if propFilesExist; then
+    echo "Copying properties files to $ycsb_dir"
+    cp *.properties $ycsb_dir
```
Comment (karianna) [inline perf/ycsb/getPropertiesFiles.sh]: what if cp fails?

File: perf/ycsb/playlist.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../TKG/playlist.xsd">
+	<include>../perf.mk</include>
+	<test>
+		<testCaseName>ycsb-azurecosmos-load</testCaseName>
+		<command>$(Q)$(TEST_RESROOT)$(D)ycsb-0.17.0/bin/ycsb.sh$(Q) load azurecosmos -P $(Q)$(TEST_RESROOT)$(D)ycsb-0.17.0/workloads/workloadc$(Q) -P $(Q)$(TEST_RESROOT)$(D)azurecosmos.properties$(Q)</command>
```
Comment (karianna) [inline perf/ycsb/playlist.xml]: same again on templating numbers etc, can we extract?

**Review by karianna — CHANGES_REQUESTED** [review]

File: perf/ycsb/README.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+YCSB is an [open-source](https://github.com/brianfrankcooper/YCSB/) database benchmark suite. AQA performance testing currently only supports YCSB with Azure Cosmos DB, but other databases may be added in the future.
+
+## Running YCSB with Azure Cosmos DB
+
+Refer to [the source](https://github.com/brianfrankcooper/YCSB/tree/master/azurecosmos) for an indepth explanation of running YCSB with Azure Cosmos DB, or follow the instructions below.
+
+## Setup
+
+This benchmark expects you to have pre-created the database "ycsb" and collection "usertable" before running the commands. When prompted for a Partition Key, use "id". For RUs, select a value you want to benchmark. RUs are the measure of provisioned thoughput that Azure Cosmos DB defines. The higher the RUs, the more throughput you will get.
```
Comment (brianjstafford) [inline perf/ycsb/README.md]: What does RU stand for, out of curiosity?

File: perf/ycsb/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Properties
+
+Configuration parameters for the benchmark should be set in the azurecosmos.properties file, which should be created in the ycsb directory (the current directory). A template version of this file azurecosmos-template.properties is included for reference.
+
+There are two required parameters for connecting to the database:
+
+azurecosmos.uri=< uri string >
+* Path to your Azure Cosmos DB account and can be obtained from the portal. It will look like the following: https://.documents.azure.com:443/
```
Comment (brianjstafford) [inline perf/ycsb/README.md]: Is this URL example accurate? (Leading "." seems wrong?)

Comment (brianjstafford) [inline perf/ycsb/README.md]: Nit: "which" instead of "and"?

File: perf/ycsb/README.md (hunk 6 above)
Comment (calderast) [reply perf/ycsb/README.md]: RUs = Request Units, described here: https://learn.microsoft.com/en-us/azure/cosmos-db/request-units - I'll add this clarifying link to the README

**Review by gdams — APPROVED** [review]
LGTM, we can follow up by extracting the version number in a separate PR. @calderast please can you raise one?

**Pull request merged**